src/codestyle.sh

```diff
@@ -0,0 +1,161 @@
+# Checkpatch is a useful tool provided by Linux, and the main goal of the code
+# in this file is to handle this script in a way to make this tool easier for
+# users.
+
+include "${KW_LIB_DIR}/lib/kw_config_loader.sh"
+include "${KW_LIB_DIR}/lib/kwlib.sh"
+
+declare -gA options_values
+
+# Runs checkpatch in the given path, which might be a file or directory.
+#
+# @FILE_OR_DIR_CHECK Target path for running checkpatch script
+function codestyle_main()
+{
+  local path
+  local flag
+  local checkpatch_options="${configurations[checkpatch_opts]}"
+  local -r original_working_dir="$PWD"
+  local kernel_root
+  local checkpatch
+  local cmd_script
+
+  parse_codestyle_options "$@"
+  if [[ "$?" != 0 ]]; then
+    complain "Invalid option: ${options_values['ERROR']}"
+    return 22 # EINVAL
+  fi
+
+  flag=${options_values['TEST_MODE']}
```

Comment:
If I understood correctly, you are using this `TEST_MODE` option passed for testing reasons. However, the options accepted in the parse functions (`kw codestyle <option>`) should be dedicated to the user, so adding this option (which the user will not use) doesn't fit well.

My suggestion it to create a function inside `codestyle_main` that is called after the parsing of options and settings of the `flag` variable, and that encapsulate all logic currently done by `codestyle_main`. This created function can then accept a `flag` argument and you can pass the value `TEST_MODE` without having to add it as an option to the parse phase.

**Obs.:** If you look throughout the project, we usually don't test the main function mainly because of this problem of parsing and setting `TEST_MODE` flags. The main function should be a "mediator" and delegate the real work for other functions, which are the actual units to test. That being said, I know you did this way because of the legacy way `checkpatch_wrapper` was implemented :grimacing:

Reply:
Oh, I understand.  We can surely do it :)

However, I think it's important to mention that some of the other kw features do this same thing when parsing options. Examples are:
- `kw ssh`
- `kw explore`
- `kw diff`
- `kw debug`
- `kw deploy`

So, wouldn't it be more right to change everything instead of just `kw codestyle`? 

Nonetheless, there is one specific line that really doesn't make sense for me now, which we copied from `kw ssh`:
```shell
  local long_options='verbose,test_mode,help'
```
I think it's unnecessary to put 'test_mode' there since the user might never execute `kw codestyle --test_mode`; so we should remove it.